otter/integration/lib/test_autoscale.py

```diff
@@ -98,3 +100,83 @@ def test_poll_until_timeout(self):
             self.assertNoResult(d)
         self.clock.advance(10)
         self.failureResultOf(d, TimedOutError)
+
+
+class GetServicenetIPs(SynchronousTestCase):
+    """
+    Tests for :func:`ScalingGroup.get_servicenet_ips`.
+    """
+    def setUp(self):
+        """Create our simulated clock and scaling group."""
+        self.clock = Clock()
+        self.pool = object()
+        self.queried_server_ids = []
+
+        class FakeRCS(object):
+            endpoints = {'nova': 'novaurl'}
+            token = "token"
+
+        @attributes(['code', 'fake_json_response'])
+        class Resp(object):
+            pass
+
+        class FakeTreq(object):
```

Comment:
I was going to use that and StubResponse, but I think @sam-falvo and @kivattik needed to use as little of the otter-specific test utilities as possible, because they needed to split the library out at some point.  Hence why find_endpoint and the auth stuff is all wrapped.